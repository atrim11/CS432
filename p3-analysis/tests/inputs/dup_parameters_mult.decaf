def int add(int x, int x, int y, int y, int z, int z) { return x; }  // Error: duplicate parameter 'x'
