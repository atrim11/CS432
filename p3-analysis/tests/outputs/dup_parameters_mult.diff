--- outputs/dup_parameters_mult.txt	2024-10-25 11:03:55.819876503 -0400
+++ expected/dup_parameters_mult.txt	2024-10-25 11:03:11.728652219 -0400
@@ -1,31 +1,4 @@
+Duplicate symbols named 'x' in scope started on line 1
+Duplicate symbols named 'y' in scope started on line 1
+Duplicate symbols named 'z' in scope started on line 1
 Program does not contain a 'main' function
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
-Duplicate parameter names 'x' in function 'add' on line 1
