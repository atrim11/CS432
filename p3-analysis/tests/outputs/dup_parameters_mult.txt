Program does not contain a 'main' function
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
