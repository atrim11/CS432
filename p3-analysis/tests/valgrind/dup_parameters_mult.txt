==4177799== Memcheck, a memory error detector
==4177799== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==4177799== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==4177799== Command: ../decaf inputs/dup_parameters_mult.decaf
==4177799== 
Program does not contain a 'main' function
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
Duplicate parameter names 'x' in function 'add' on line 1
==4177799== 
==4177799== HEAP SUMMARY:
==4177799==     in use at exit: 0 bytes in 0 blocks
==4177799==   total heap usage: 1,810 allocs, 1,810 frees, 216,900 bytes allocated
==4177799== 
==4177799== All heap blocks were freed -- no leaks are possible
==4177799== 
==4177799== For lists of detected and suppressed errors, rerun with: -s
==4177799== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
