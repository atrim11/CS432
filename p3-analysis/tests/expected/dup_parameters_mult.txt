Duplicate symbols named 'x' in scope started on line 1
Duplicate symbols named 'y' in scope started on line 1
Duplicate symbols named 'z' in scope started on line 1
Program does not contain a 'main' function
